src/dms/core/EdFi.DataManagementService.Core/Backend/UpdateCascadeHandler.cs

```diff
@@ -7,80 +7,102 @@
 using System.Text.Json;
 using System.Text.Json.Nodes;
 using EdFi.DataManagementService.Core.ApiSchema;
-using EdFi.DataManagementService.Core.ApiSchema.Extensions;
+using EdFi.DataManagementService.Core.ApiSchema.Helpers;
 using EdFi.DataManagementService.Core.External.Backend;
+using EdFi.DataManagementService.Core.External.Model;
 using Json.More;
 using Json.Path;
 using Microsoft.Extensions.Logging;
+using JsonPath = Json.Path.JsonPath;
 
 namespace EdFi.DataManagementService.Core.Backend;
 
 public class UpdateCascadeHandler(IApiSchemaProvider _apiSchemaProvider, ILogger _logger)
     : IUpdateCascadeHandler
 {
+    private static JsonNode? FindResourceNode(
+        ApiSchemaDocuments apiSchemaDocuments,
+        ProjectName projectName,
+        ResourceName resourceName
+    )
+    {
+        ProjectSchema? projectSchema = apiSchemaDocuments.FindProjectSchemaForProjectName(projectName);
+        if (projectSchema == null)
+        {
+            return null;
+        }
+
+        return projectSchema.FindResourceSchemaNodeByResourceName(resourceName);
+    }
+
     public UpdateCascadeResult Cascade(
         JsonElement originalEdFiDoc,
-        string originalDocumentProjectName,
-        string originalDocumentResourceName,
+        ProjectName originalDocumentProjectName,
+        ResourceName originalDocumentResourceName,
         JsonNode modifiedEdFiDoc,
         JsonNode referencingEdFiDoc,
         long referencingDocumentId,
         short referencingDocumentPartitionKey,
         Guid referencingDocumentUuid,
-        string referencingProjectName,
-        string referencingResourceName
+        ProjectName referencingProjectName,
+        ResourceName referencingResourceName
     )
     {
         var isIdentityUpdate = false;
         JsonNode returnEdFiDoc = referencingEdFiDoc.DeepClone();
 
-        var apiSchemaDocument = new ApiSchemaDocument(_apiSchemaProvider.CoreApiSchemaRootNode, _logger);
+        ApiSchemaDocuments apiSchemaDocuments = new(
+            _apiSchemaProvider.GetApiSchemaNodes(),
+            _logger
+        );
 
-        var originalResourceNode =
-            apiSchemaDocument.FindResourceNode(originalDocumentProjectName, originalDocumentResourceName)
+        JsonNode originalResourceNode =
```

Comment:
Took a lot of un-`var`ing to figure out bug of mixing endpoint and resource names